codeit/services.js/ArticleService.js

```diff
@@ -0,0 +1,125 @@
+const BASE_URL = 'https://panda-market-api-crud.vercel.app'; // 기본이 되는 URL
+
+const ArticleService = { 
+    
+// 1. 게시글 목록 조회 :  getArticleList 함수
+    getArticleList(params = {page : 1, pageSize: 10, keyword: ''} ) { // 이 함수를 호출할 때, 어떤 조건으로 보여줄지 기본값& 파라미터를 설정함
+        const url = new URL(`${BASE_URL}/articles`); // 기본 url에 path 주소를 추가함
+
+        Object.keys(params) // 객체로 되어있는 params의 프로퍼티 네임(키)들을 배열로 만듦
+            .forEach((key)=>{ // 배열로 만든 params의 키들을 하나씩 꺼내서 콜백 함수 실행
+                if (params[key] !== undefined && params[key] !== '') { // params[key] 는 해당 키의 값 -> 값이 undefined, 빈 문자열이 아닐경우에만 동작
+                    url.searchParams.append(key, params[key]); // new URL()로 만든 URL 객체 뒤에 파라미터에 들어가 있는 키와 값을 한 쌍의 쿼리 파라미터를 추가해주는 메서드
```

Comment:
중복 파라미터가 없기 때문에 `append`보다는 `set`이 더 어울립니다.
`append`는 동일한 키를 여러 번 추가할 때 사용하는데, 현재 API 스펙상 한 키는 한 값만 전달됩니다.
기능적으로는 문제 없지만, 코드를 읽는 개발자 입장에서는 의도를 명확히 드러내는 `set`이 더 적합합니다.

```suggestion
                    url.searchParams.set(key, params[key]); // new URL()로 만든 URL 객체 뒤에 파라미터에 들어가 있는 키와 값을 한 쌍의 쿼리 파라미터를 추가해주는 메서드
```